integration/flink/src/main/java/io/openlineage/flink/OpenLineageFlinkJobListener.java

```diff
@@ -1,116 +1,146 @@
 package io.openlineage.flink;
 
+import io.openlineage.client.OpenLineage;
+import io.openlineage.flink.api.OpenLineageContext;
+import io.openlineage.flink.visitor.Visitor;
+import io.openlineage.flink.visitor.VisitorFactory;
+import io.openlineage.flink.visitor.VisitorFactoryImpl;
+import java.lang.reflect.Field;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.lang3.reflect.FieldUtils;
 import org.apache.flink.api.common.JobExecutionResult;
 import org.apache.flink.api.dag.Transformation;
-import org.apache.flink.connector.kafka.sink.KafkaRecordSerializationSchema;
-import org.apache.flink.connector.kafka.sink.KafkaSink;
-import org.apache.flink.connector.kafka.source.KafkaSource;
-import org.apache.flink.connector.kafka.source.enumerator.subscriber.KafkaSubscriber;
 import org.apache.flink.core.execution.JobClient;
 import org.apache.flink.core.execution.JobListener;
 import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
-import org.apache.flink.streaming.api.transformations.OneInputTransformation;
-import org.apache.flink.streaming.api.transformations.PartitionTransformation;
-import org.apache.flink.streaming.api.transformations.SinkTransformation;
-import org.apache.flink.streaming.api.transformations.SourceTransformation;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import javax.annotation.Nullable;
-import java.lang.reflect.Field;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
-import java.util.function.Function;
 
+@Slf4j
 public class OpenLineageFlinkJobListener implements JobListener {
 
-    private final StreamExecutionEnvironment executionEnvironment;
-    private static final Logger LOGGER = LoggerFactory.getLogger(OpenLineageFlinkJobListener.class);
+  private final StreamExecutionEnvironment executionEnvironment;
+  private final OpenLineageContext openLineageContext;
 
+  public OpenLineageFlinkJobListener(StreamExecutionEnvironment executionEnvironment) {
+    makeTransformationsUnclearableList(executionEnvironment);
+    this.executionEnvironment = executionEnvironment;
+    this.openLineageContext =
+        OpenLineageContext.builder()
+            .streamExecutionEnvironment(Optional.of(executionEnvironment))
+            .openLineage(new OpenLineage(URI.create("")))
```

Comment:
I'm working on this URI part, trying to move code from spark side, I will think about making it common for both Flink/Spark modules